### DISCUSSION TITLE
Request for feedback: Avalonia Learn and Developer Certification

### DISCUSSION THREAD

**Discussion by MikeCodesDotNET** [AvaloniaUI/Avalonia#16420] [Ideas]
I'm eager to share an idea we've been working on to help developers learn Avalonia and demonstrate their skills. We'd love to hear your thoughts on this initiative.

### Avalonia Learn
Avalonia Learn will be an in-depth educational video series designed to teach developers everything they need to know to be successful with Avalonia. These videos will be:

- Hosted on our customer portal
- Accessible to subscribers
- Regularly updated to provide the latest information

Our goal is to create exceptional learning materials that cover a wide range of topics, from basics to advanced concepts, ensuring that developers at all levels can benefit.

### Developer Certification
Alongside Avalonia Learn, we're planning to introduce a developer certification program. This certification will:

- Be included in the Learn subscription
- Be available separately for those who already consider themselves experts
- Validate a developer's Avalonia skills

**The certification process will involve:**

- A 2-hour, multiple-choice test
- 200 questions that progressively increase in difficulty
- Up to three attempts to pass
- A passing score of at least 80% correct answers

Upon successful certification, developers will have the option to:

- Download a digital certificate or order a physical copy (for a modest additional fee, depending on the overall price). 
- Be featured in a dedicated section of our website
- Mark themselves as "open to work" to attract potential work opportunities

Certifications will be valid for one year, with optional renewals to ensure developers stay up-to-date with the latest Avalonia developments. 

Below is a WIP of the exam feature within our customer portal. 
<img width="1145" alt="DevCert_Portal" src="https://github.com/user-attachments/assets/d989d87e-f066-4a4e-8beb-8d12b4519a70">

### Addressing the Elephant in the Room: Pricing

We anticipate that some community members might suggest that Avalonia Learn should be free to help grow the Avalonia community. While we absolutely understand and appreciate this perspective, we want to be transparent about the realities of creating such a program.

Developing and maintaining high-quality learning materials requires a significant investment of time, expertise, and resources. To ensure we can deliver and continuously improve these materials, we need to implement a pricing model that at least allows the program to break even.

We're committed to being as inclusive as possible. We plan to offer complementary access in various situations and are exploring ways to make the program accessible to as many developers as possible. 

However, to make this initiative sustainable and ensure its long-term success, we cannot offer it entirely for free. We face the choice of either not creating this comprehensive learning program at all or implementing a modest cost to sustain it. A high-quality, continually updated learning resource will provide immense value to the Avalonia community and is worth this investment.

### We Need Your Input 
This is where we really need your help. We want to find a pricing model that balances accessibility with sustainability. Your feedback is crucial in helping us determine the right approach.

Please take a moment to fill out our [Google Form](https://docs.google.com/forms/d/e/1FAIpQLScXX4EiliiWQX0RmzZCFnQP7DauiQF61EZV-LN0_qT2IHmySg/viewform) with your thoughts on pricing and other aspects of this initiative. Your input will directly influence how we structure and price Avalonia Learn and the certification program.

Some key areas we're particularly interested in:

- What price range would you consider reasonable for these services?
- How would you value the certification separately from the learning materials?
- What additional features or benefits would increase the value for you?

Your honest feedback is invaluable in helping us create a program that genuinely serves the Avalonia community while ensuring its long-term viability. We're committed to providing high-quality educational resources and meaningful ways for developers to showcase their expertise, and your input will help us achieve this goal.

Thank you for your time, understanding, and ongoing support of Avalonia. We're looking forward to reading your responses and discussing this further with the community!

**Comment by ai-van-han** [comment]
why not follow the Umbraco Pricing?

**Comment by LaurentInSeattle** [comment]
Learn: I tried the online questionnaire, even if not really interested, it tried to force me to enter non zero prices, so I left the page. 

Cert: For years I have not seen "Avalonia" mentioned in any JD requirements or "nice to have" skills on LinkedIn or Indeed. So I doubt that a Certification would ever be useful to clinch even a job interview... If on top of that, I would have to pay for it, of course I pass.
